File: LearningSystem.Core/Entities/Lecture.cs
namespace LearningSystem.Core.Entities
{
    using System.Collections.Generic;

    public class Lecture : AuditInfo, IEntity
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public int CourseId { get; set; }

        public Course Course { get; set; }

        public string VideoUrl { get; set; }

        public List<HomeworkAssignment> HomeworkAssignments { get; set; }
    }
}

File: LearningSystem.Web/ClientApp/src/app/_models/lecture.ts
import { HomeworkAssignment } from "./homeworkAssignment";
import { SafeResourceUrl } from "@angular/platform-browser";

export class Lecture {
  id: number;
  name: string;
  description: string;
  videoUrl: string;
  safeVideoUrl: SafeResourceUrl;
  homeworkAssignments: HomeworkAssignment[];
}

export class LectureInput {
  id: number;
  name: string;
  description: string;
  videoUrl: string;
  courseId: number;
}
